e2e/e2etests/test_message_passing_zevm.go

```diff
@@ -0,0 +1,332 @@
+	}
+
+	// Set destination details
+	EVMChainID, err := r.EVMClient.ChainID(r.Ctx)
+	if err != nil {
+		panic(err)
+	}
+	destinationAddress := r.EvmTestDAppAddr
+
+	// Contract call originates from ZEVM chain
+	r.ZEVMAuth.Value = amount
+	tx, err := r.WZeta.Deposit(r.ZEVMAuth)
+	if err != nil {
+		panic(err)
+	}
+	r.ZEVMAuth.Value = big.NewInt(0)
+	r.Logger.Info("wzeta deposit tx hash: %s", tx.Hash().Hex())
+	receipt := utils.MustWaitForTxReceipt(r.Ctx, r.ZEVMClient, tx, r.Logger, r.ReceiptTimeout)
+	r.Logger.EVMReceipt(*receipt, "wzeta deposit")
+	if receipt.Status == 0 {
+		panic("deposit failed")
+	}
+	tx, err = r.WZeta.Approve(r.ZEVMAuth, r.ZevmTestDAppAddr, amount)
+	if err != nil {
+		panic(err)
+	}
+	r.Logger.Info("wzeta approve tx hash: %s", tx.Hash().Hex())
+	receipt = utils.MustWaitForTxReceipt(r.Ctx, r.ZEVMClient, tx, r.Logger, r.ReceiptTimeout)
+	r.Logger.EVMReceipt(*receipt, "wzeta approve")
+	if receipt.Status == 0 {
+		panic(fmt.Sprintf("approve failed, logs: %+v", receipt.Logs))
+	}
+	testDAppZEVM, err := testdapp.NewTestDApp(r.ZevmTestDAppAddr, r.ZEVMClient)
+	if err != nil {
+		panic(err)
+	}
+	// Call the SendHelloWorld function on the ZEVM dapp Contract which would in turn create a new send, to be picked up by the zetanode evm hooks
+	// set Do revert to true which adds a message to signal the EVM zetaReceiver to revert the transaction
+	tx, err = testDAppZEVM.SendHelloWorld(r.ZEVMAuth, destinationAddress, EVMChainID, amount, true)
+	if err != nil {
+		panic(err)
+	}
+	r.Logger.Print("TestDApp ZEVM address: %s", r.ZevmTestDAppAddr.String())
+	r.Logger.Info("TestDApp.SendHello tx hash: %s", tx.Hash().Hex())
+	receipt = utils.MustWaitForTxReceipt(r.Ctx, r.ZEVMClient, tx, r.Logger, r.ReceiptTimeout)
+	if receipt.Status == 0 {
+		panic(fmt.Sprintf("send failed, logs: %+v", receipt.Logs))
+	}
+
+	// New inbound message picked up by zetanode evm hooks and processed directly to initiate a contract call on EVM which would revert the transaction
+	cctx := utils.WaitCctxMinedByInTxHash(r.Ctx, receipt.TxHash.String(), r.CctxClient, r.Logger, r.CctxTimeout)
+	if cctx.CctxStatus.Status != cctxtypes.CctxStatus_Reverted {
+		panic("expected cctx to be reverted")
+	}
+	r.Logger.Print(fmt.Sprintf("🔄 Cctx mined for contract call chain zevm %s", cctx.Index))
+
+	// On finalization the Fungible module calls the onRevert function which in turn calls the onZetaRevert function on the sender contract
+	receipt, err = r.ZEVMClient.TransactionReceipt(r.Ctx, ethcommon.HexToHash(cctx.GetCurrentOutTxParam().OutboundTxHash))
+	if err != nil {
+		panic(err)
+	}
+	if receipt.Status != 1 {
+		panic("tx failed")
+	}
+	receivedHelloWorldEvent := false
+	for _, log := range receipt.Logs {
+		_, err := testDAppZEVM.ParseRevertedHelloWorldEvent(*log)
+		if err == nil {
+			r.Logger.Info("Received HelloWorld event:")
+			receivedHelloWorldEvent = true
+		}
+	}
+	if !receivedHelloWorldEvent {
+		panic("expected HelloWorld event")
```

Comment:
For these ones maybe we can print the rawLogs in the panic

Reply:
d895bb494c3f8b659de6508e4307d4770c6d05f8